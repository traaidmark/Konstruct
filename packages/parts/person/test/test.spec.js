// PART [ PERSON > TESTS ] #####################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Person from '../src';

// 1.2. END ....................................................................

const args = {
  name: 'Aubrey Turner',
  role: 'Chief Growth Officer',
  imgSrc: 'https://s3.amazonaws.com/cdn.wealthpoint.app/person_aubrey_f6ce6120a8.jpg',
  className: '',
};

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Part: Person', () => {

  // 2.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Person />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 2.1. END ..................................................................

  // 2.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Person { ...args } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 2.2. END ..................................................................

});

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
