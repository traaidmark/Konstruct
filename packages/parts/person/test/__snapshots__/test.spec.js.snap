// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Part: Person renders correctly with all props 1`] = `
<aside
  className="p-person "
>
  <figure>
    <img
      alt="Aubrey Turner"
      src="https://s3.amazonaws.com/cdn.wealthpoint.app/person_aubrey_f6ce6120a8.jpg"
    />
  </figure>
  <strong>
    Aubrey Turner
  </strong>
  <small>
    Chief Growth Officer
  </small>
</aside>
`;

exports[`Part: Person renders correctly without any props 1`] = `
<aside
  className="p-person "
/>
`;
